### ISSUE TITLE
Rubocop uses incorrect indentation for new `end` if it needs to correct indentation and add an `end`.

### ISSUE THREAD

**Issue by Lykos** [rubocop-hq/rubocop#7706]
There are interactions between two corrections that lead to incorrect indentation. If the indentation has to be corrected and at the same time it also has to add a line with an `end`, it will add that to the previous broken indentation level.

It's not a big deal, but it can be annoying (but if you have limited time, my other bug I reported is certainly more severe).

--------

## Expected behavior

Rubocop adds new `end` tags to the correct indentation level which is the one of the surrounding code _after_ that one got its indentation correction applied.

## Actual behavior

Rubocop adds new `end` tags to the incorrect indentation level which is the one of the surrounding code _before_ that one got its indentation correction applied.

## Steps to reproduce the problem

Add the following file:

```
module Stuff
def a(input)
  puts 'some string that is long enough so our line violates the line limit' if input == 2
  puts 2
end
end
```

Now Rubocop wants to do two corrections (okay, it wants to do more than 2, but 2 that we care about):

- Correct the indentation of `def a(input)`
- Change the modifier conditional `if input == 2` to a normal conditional.

The changes seem to interact in such a way that the resulting `end` is incorrectly indented.

## RuboCop version

0.79.0 (using Parser 2.7.0.2, running on ruby 2.5.1 x86_64-linux-gnu)

**Comment by koic** [comment]
`AutoCorrect` of `Layout/EndAlignment` cop is disabled by default.
https://github.com/rubocop-hq/rubocop/blob/v0.79.0/config/default.yml#L505

Please add the following to your .rubocop.yml:

```yaml
# .rubocop.yml
Layout/EndAlignment:
  AutoCorrect: true
```

Note: This is a frequent issue, and is the same as #7061.
